### DISCUSSION TITLE
How to addFilteredStreamRule?

### DISCUSSION THREAD

**Discussion by nniiggeell** [redouane59/twittered#204] [Q&A]
Hi, I tried adding a filtered stream rule. "@mentions" but I get an illegalArgumentException. When I try to follow the test showed in the ITwitterClientV2Test file, I also get the IAE. What am I doing wrong?

![image](https://user-images.githubusercontent.com/77267877/120164579-4f9df180-c22d-11eb-91f5-4a1a06499a6f.png)
![image](https://user-images.githubusercontent.com/77267877/120164629-59bff000-c22d-11eb-86a4-890772f2e55d.png)

**Comment by redouane59** [comment]
Hi,
Can you debug inside the `addFilteredStreamRule` to see in which line the exception is thrown ?

**Reply by redouane59** [reply]
It's possible that you are trying to create a rule already created.

**Reply by nniiggeell** [reply]
Hi, thanks for the reply. You were right, I was trying to add rules with the same tag/value. 
However, now that I have experimented, I realised that I have added too many StreamRules and I want to remove them all. This is my code but it does not work.
![image](https://user-images.githubusercontent.com/77267877/120261553-b2919600-c2ca-11eb-9b72-658c4eada4c7.png)

And this is my output
![image](https://user-images.githubusercontent.com/77267877/120261593-c63cfc80-c2ca-11eb-8c8c-80cf44c52f09.png)

I get back the size of the list but nothing else runs even after 2 minutes

**Reply by Greg-M73** [reply]
Hi @nniiggeell ,

You have to use deleteFilteredStreamRuleTag().
Here is my code to clear existing filter :
![image](https://user-images.githubusercontent.com/37636873/120300909-aa3a5a80-c2cc-11eb-9e5a-0ecd8291d0cd.png)

**Reply by nniiggeell** [reply]
Thank you @Greg-M73 ! This worked!
